src/value.cpp

```diff
@@ -449,16 +449,38 @@ expr Tensor::to1DArrayWithOfs(
 
 MemRef::MemRef(Memory *m): m(m), bid(ctx), offset(ctx), layout(Layout({}, ctx)) {}
 
+MemRef::MemRef(Memory *m,
+    const std::vector<expr> &dims,
+    const Layout &layout,
+    const z3::sort &elemty,
+    bool freshBlock): m(m), bid(ctx), offset(ctx), dims(dims), layout(layout) {
+  if (freshBlock) {
+    bid = m->addLocalMemBlock(get1DSize());
+    offset = Index::zero();
+  } else {
+    static int count = 0;
+    string freshName =  + "memref" + to_string(count ++);
+    bid = ctx.bv_const((freshName + "_bid").c_str(), m->getBIDBits());
+    offset = Index((freshName + "_offset").c_str());
+  }
+}
+
```

Comment:
always create new fresh name for "bid", "offset" when memref name is not specified.